drgn/helpers/linux/tcp_states.py

```diff
@@ -0,0 +1,54 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+"""
+TCP States
+----------
+
+The ``drgn.helpers.linux.tcp_states`` module provides definitions for the TCP
+protocol sk_state field defined in :linux:`include/net/tcp_states.h`.
+"""
+
+__all__ = [
+    'TcpStates'
+]
+
+
+class TcpStates(object):
+    """
+    Definitions for the TCP protocol sk_state field.
+    """
+
+    ESTABLISHED = 1
+    SYN_SENT = 2
+    SYN_RECV = 3
+    FIN_WAIT1 = 4
+    FIN_WAIT2 = 5
+    TIME_WAIT = 6
+    CLOSE = 7
+    CLOSE_WAIT = 8
+    LAST_ACK = 9
+    LISTEN = 10
+    CLOSING = 11
+    NEW_SYN_RECV = 12
```

Comment:
Are these values guaranteed not to change? If not, then we should grab them from the debug info instead of hard-coding. E.g., something like this:

```python
import enum


def get_tcp_states(prog):
    try:
        return prog.cache['TcpStates']
    except KeyError:
        pass
    # The enum type is anonymous, but we can get it from one of the
    # enumerators.
    tcp_states_type = prog['TCP_ESTABLISHED'].type_.enumerators
    enumerators = [
        (name[4:] if name.startswith('TCP_') else name, value)
        for (name, value) in tcp_states_type
        if name != 'TCP_MAX_STATES'
    ]
    TcpStates = enum.IntEnum('TcpStates', enumerators)
    prog.cache['TcpStates'] = TcpStates
    return TcpStates
```

Reply:
Oh, wow.

TIL: that it's possible to get a `enum` from `prog`
TIL: `prog.cache`

These values should never change, since e.g. glibc header `<netinet/tcp.h>` and kernel's `include/uapi/linux/bpf.h` use same enumerations, but new states (even though just one was added since 2005) may appear with delay in new places, e.g. the last enumeration `TCP_NEW_SYN_RECV` was introduced in 2015 but it's not present on my devbox in `<netinet/tcp.h>`. 

The only reason I defined them manually is I didn't know how to get them automatically and getting them automatically is much better IMO since for example it avoids possible problems with "recently" introduced  `TCP_NEW_SYN_RECV`.

I'll switch to your version. Thanks!

Reply:
While switching to the automatic enum generation I realized that it would require changing all interfaces that uses TCP states to add additional `prog` argument (e.g. `sk_fullsock`).

That's probably not a big deal, but I'm curious if it's possible to get access to `prog` variable w/o passing it as an argument to helper. The benefit would be have helpers with signatures similar to those in kernel (e.g. same `sk_fullsock`)

Reply:
As long as the helper takes a `drgn.Object`, you can grab it from `obj.prog_` (e.g., `sk.prog_` for `sk_fullsock`).